framework/src/auxkernels/MaterialStdVectorAux.C

```diff
@@ -15,27 +15,24 @@
 #include "MaterialStdVectorAux.h"
 
 template<>
-InputParameters validParams<MaterialStdVectorAux>()
+InputParameters validParams< MaterialStdVectorAux>()
 {
-  InputParameters params = validParams<MaterialAuxBase<std::vector<Real> > >();
-  params.addParam<unsigned int>("index", 0, "The index to consider for this kernel");
-  params.addClassDescription("Extracts a component of a material's std::vector<Real> to an aux variable.  If the std::vector is not of sufficient size then zero is returned");
-
+  InputParameters params = validParams<MaterialStdVectorAuxBase<> >();
+  params.addClassDescription("Extracts a component of a material type std::vector<Real> to an aux variable.  If the std::vector is not of sufficient size then zero is returned");
   return params;
 }
 
-MaterialStdVectorAux::MaterialStdVectorAux(const std::string & name, InputParameters parameters) :
-  MaterialAuxBase<std::vector<Real> >(name, parameters),
-    _index(getParam<unsigned int>("index"))
+ MaterialStdVectorAux:: MaterialStdVectorAux(const std::string & name, InputParameters parameters) :
+  MaterialStdVectorAuxBase<Real>(name, parameters)
 {
 }
 
-MaterialStdVectorAux::~MaterialStdVectorAux()
+ MaterialStdVectorAux::~ MaterialStdVectorAux()
 {
 }
 
 Real
-MaterialStdVectorAux::computeValue()
+ MaterialStdVectorAux::computeValue()
 {
   mooseAssert(_prop[_qp].size() > _index, "MaterialStdVectorAux: You chose to extract component " << _index << " but your Material property only has size " << _prop[_qp].size());
   return _factor * _prop[_qp][_index] + _offset;
```

Comment:
ComputeComponent would be relevant to only vectors and tensors. I could probably name it as computeMaterialValue() and call it within computeValue of the MaterialAux base class.